auth/auth.go

```diff
@@ -749,11 +773,23 @@ func (auth *Authenticator) authenticateOIDCIdentity(identity *Identity, provider
 		user, err = auth.RegisterNewUser(username, identity.Email)
 		if err != nil && !base.IsCasMismatch(err) {
 			base.DebugfCtx(auth.LogCtx, base.KeyAuth, "Error registering new user: %v", err)
-			return nil, time.Time{}, err
+			return nil, PrincipalConfig{}, time.Time{}, err
+		}
+	}
+
+	if user != nil {
+		now := time.Now()
+		updates = PrincipalConfig{
+			Name:            base.StringPtr(user.Name()),
+			Email:           &identity.Email,
+			OIDCIssuer:      &provider.Issuer,
+			OIDCRoles:       oidcRoles,
+			OIDCChannels:    oidcChannels,
+			OIDCLastUpdated: &now,
```

Comment:
Does OIDCLastUpdated have a functional purpose in the context of this PR, or is it in place for a future enhancement?

Reply:
It's presented through the admin API (`/<db>/_user/<user>`) - it was the compromise we decided to avoid confusing admins when they see users with stale OIDC roels/channels.